### ISSUE TITLE
 BadParametersError in CircleCI OS X machine (locally runs fine)

### ISSUE THREAD

**Issue by DmitrijKotsur** [appium/appium#11392]
## The problem
Test failure with `Encountered internal error running command: BadParametersError: Parameters were incorrect. We wanted "W3C protocol expects any of script, pageLoad or implicit to be set" and you sent {"type":"implicit","ms":20000}`, but locally everything works fine.

### **CircleCI** machine:
* Appium version: 1.9.0
* Desktop OS/version used to run Appium: OS X 10.13.3
* Node.js version: v10.4.1
* Mobile platform/version under test: iOS
* Real device or emulator/simulator: simulator
### **Local** machine:
* Appium version: 1.8.1
* Desktop OS/version used to run Appium: OS X 10.13.6
* Node.js version: v9.5.0
* Mobile platform/version under test: iOS
* Real device or emulator/simulator: simulator
## Details
App starts on remote simulator, because I have on-failure-screenshots with start screen.
Test doesn't runs in parallel.
## Link to Appium logs
https://gist.github.com/DmitrijKotsur/727994918c3cfbbf7eae9dcea572881d#file-appium-test-logs
## Code To Reproduce Issue [ Good To Have ]
https://gist.github.com/DmitrijKotsur/727994918c3cfbbf7eae9dcea572881d#file-appiumdrivercontroller-start-method

**Comment by mykola-mokhnach** [comment]
This is fixed in appium@beta

**Comment by DmitrijKotsur** [comment]
> This is fixed in appium@beta

Thank you for answer!
I fix my problem with appium@1.8.1, because appium@beta skipped one of my test method. 
I can create new ticket for problem in appium@beta if you want.

**Comment by mykola-mokhnach** [comment]
What is the problem there? Can you attach the log?

**Comment by DmitrijKotsur** [comment]
> What is the problem there? Can you attach the log?

With great pleasure!
Logs v.1.8.1 (all right)
https://gist.github.com/DmitrijKotsur/727994918c3cfbbf7eae9dcea572881d#file-good-logs-appium-1-8-1
Logs of the beta
https://gist.github.com/DmitrijKotsur/727994918c3cfbbf7eae9dcea572881d#file-bad-logs-appium-beta
Test:
https://gist.github.com/DmitrijKotsur/727994918c3cfbbf7eae9dcea572881d#file-timerbuttonstest-method
Method with failure:
https://gist.github.com/DmitrijKotsur/727994918c3cfbbf7eae9dcea572881d#file-method-with-failure

**Comment by mykola-mokhnach** [comment]
@DmitrijKotsur Are you sure your java client is up to date?

**Comment by DmitrijKotsur** [comment]
I use v6.0.0. Is it wrong?

**Comment by mykola-mokhnach** [comment]
It does not look like it... The client detects the protocol as OSS

Can you attach the whole server log?

**Comment by DmitrijKotsur** [comment]
@mykola-mokhnach 
`INFO: Detected dialect: W3C` - look at https://gist.github.com/DmitrijKotsur/727994918c3cfbbf7eae9dcea572881d#file-bad-logs-appium-beta - it's log of test running with appium v.1.9.1.beta-2. OSS detected with 1.8.1
